updatecli/scripts/check-jdk.sh

```diff
@@ -52,7 +52,8 @@ case "${1}" in
   19.*+*)
     platforms=("x64_linux" "x64_windows" "aarch64_linux" "s390x_linux");;
   21*+*)
-    platforms=("x64_linux" "x64_windows" "aarch64_linux" "s390x_linux");;
+    # No s390x support for JDK21 yet
+    platforms=("x64_linux" "x64_windows" "aarch64_linux");;
```

Comment:
What happens when there is a JDK upgrade not available with the s390x (such as `21+35-ea` => `21+35`)?

- Updatecli still open a PR with the new version in the parameters files (hieradata)
- The build passes as this is only a "hieradata -> puppet -> JCasc YAML file" check
- Then ci.jenkins.io gets updated when the updatecli PR is merged and its "tool" are setup with the new version
- And then, all builds calling the `tool 'jdk21'` pipeline instructions fails as the tool installer responds a HTTP/404 error.


There is no problem on the top-level idea to stop (temporarily) tracking the `s390x` and keep it to a lower version, but it means you must update the tool configuration to keep the old version with a TODO comment as part of the PR here.

Reply:
yes but this mean changing the target to point on each of them no ? 
`key: $.profile::jenkinscontroller::jcasc.tools_default_versions.jdk21`

Reply:
That would work yes. Or you can also keep the updatecli code as it to track and define the default JDK21 version (e.g. `21+35`) and define an exception in the specific hieradata of ci.jenkins.io which defines the s390x agent, as the override mechanism is for the most specific cases like this one.

Reply:
For info, see https://github.com/jenkins-infra/jenkins-infra/pull/3113#issuecomment-1786727093